vault/data_source_azure_access_credentials.go

```diff
@@ -0,0 +1,243 @@
+				Required:    true,
+				Description: "Azure Secret Backend to read credentials from.",
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Azure Secret Role to read credentials from.",
+			},
+			"validate_creds": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether generated credentials should be validated before being returned.",
+				Default:     false,
+			},
+			"num_sequential_successes": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     8,
+				Description: `If 'validate_creds' is true, the number of sequential successes required to validate generated credentials.`,
+			},
+			"num_seconds_between_tests": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     7,
+				Description: `If 'validate_creds' is true, the number of seconds to wait between each test of generated credentials.`,
+			},
+			"max_cred_validation_seconds": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     20 * 60, // 20 minutes
+				Description: `If 'validate_creds' is true, the number of seconds after which to give up validating credentials.`,
+			},
+			"subscription_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `If 'validate_creds' is true, the subscription id to use to test whether the client ID and secret are valid.`,
+			},
+			"tenant_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `If 'validate_creds' is true, the tenant id  to use to test whether the client ID and secret are valid.`,
+			},
+			"client_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The client id for credentials to query the Azure APIs.",
+			},
+			"client_secret": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The client secret for credentials to query the Azure APIs.",
+			},
+			"lease_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Lease identifier assigned by vault.",
+			},
+			"lease_duration": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Lease duration in seconds relative to the time in lease_start_time.",
+			},
+			"lease_start_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Time at which the lease was read, using the clock of the system where Terraform was running",
+			},
+			"lease_renewable": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if the duration of this lease can be extended through renewal.",
+			},
+		},
+	}
+}
+
+func azureAccessCredentialsDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	role := d.Get("role").(string)
+	path := backend + "/creds/" + role
+
+	data := make(map[string][]string)
```

Comment:
Is this needed? Can we use `client.Logical().Read()` instead?

Reply:
Ah! Yes, that was copy pasta. Updating.

Reply:
https://github.com/terraform-providers/terraform-provider-vault/pull/713/commits/c2525e132a31c199cc614fc101220373b1b7e5e0